### ISSUE TITLE
Can't sign in to Amazon

### ISSUE THREAD

**Issue by welovewebs** [Sandmann79/xbmc#608]
Thanks for the plugin!

I logged in on a FireTV device downstairs, but can't log in on this Windows 10 device (I can log in with the Windows 10 Prime Video app).
My Amazon account has 2FA set up (I have also tried without and with a different Amazon account that doesn't have 2FA).
I'm using an Android authenticator app for my OTP.

The plugin requests my OTP after the username and password, but afterwards says 'Unknown Error, please enable advanced debugging in settings and check avod-login.log/avod-login-si.log/avod-login-mfa.log'

I changed the file extension for the logfiles to .html and they render the following:

avod-login.log shows the Amazon Two-Step Verification OTP entry form (if I enter my OTP it redirects to a page telling me I must enable cookies)

avod-login-si.log shows the Amazon We're Sorry page, stating 'An error occurred when we tried to process your request. We're working on the problem and expect to resolve it shortly. Please note that if you were trying to place an order, it will not have  been processed at this time. Please try again later. We apologise for the inconvenience'

avod-login-mfa.log shows 'We apologise for the inconvenience. An error occurred when we tried to process your request'

Logfiles are attached also.

I also tried changing the region from Auto to United Kingdom which results in the same problem.
After reading another issue raised here, I tried changing the region to ROE Europe, which lets me log in successfully (including with OTP), however when I then launch the plugin, Kodi informs the following:
'PrimeVideo error
You might be unable to access the service: check the website www.primevideo.com for more information'

If I try to access www.primevideo.com from a browser in Windows, it redirects me to amazon.co.uk

Errors from kodi.log are:
CGUIMediaWindow::GetDirectory(plugin://plugin.video.amazon-test/) failed
XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.amazon-test/
Control 55 in window 10025 has been asked to focus, but it can't
[Amazon VOD] Unable to parse the navigation menu for primevideo.

[avod-login.log](https://github.com/Sandmann79/xbmc/files/7966276/avod-login.log)
[avod-login-mfa.log](https://github.com/Sandmann79/xbmc/files/7966277/avod-login-mfa.log)
[avod-login-si.log](https://github.com/Sandmann79/xbmc/files/7966278/avod-login-si.log)
[kodi.log](https://github.com/Sandmann79/xbmc/files/7966279/kodi.log)

Is there anything I can try to resolve this?

**Comment by ErnstNeger** [comment]
You seem to have the same/similar problem I do (see #602). Which python version are you using?

**Comment by Atomisirsi** [comment]
I have the same problems login into Amazon.de from Germany.
After the last login step the respective `avod-login*.log` contains a line complaining about `In unserem System ist ein interner Fehler aufgetreten.` (internal error). I have tried without MFA and also switched between the two available MFA variants.

Iam running latest CoreELEC 19.4_rc2 on an Odroid N2+.
The used Python Interpreter has version 3.8.11.

How can I support on this topic?

**Comment by ErnstNeger** [comment]
I don't think you can support, but try the latest [beta version](https://www.kodinerds.net/index.php/Thread/44211-Release-Amazon-Prime-Instant-Video/?postID=672168#post672168) from kodinerds, it might resolve your login issue.

**Comment by Atomisirsi** [comment]
> I don't think you can support, but try the latest [beta version](https://www.kodinerds.net/index.php/Thread/44211-Release-Amazon-Prime-Instant-Video/?postID=672168#post672168) from kodinerds, it might resolve your login issue.

Indeed! The 0.9.5-beta2 solved the issue. Thank's for your hint!

**Comment by stephencpd** [comment]
> You seem to have the same/similar problem I do (see #602). Which python version are you using?

Hi,

Sorry for the delay in responding. I'm using Matrix, so Python 3
I'll try the 0.9.5-beta2

**Comment by stephencpd** [comment]
0.9.5-beta2 works for me also :]

**Comment by ktaragorn** [comment]
0.9.5-beta2 got me past the initial error to the captha image, but after that hit another error..
kodi.log has this
```

2022-02-18 21:55:19.564 T:11219    INFO <general>: [Amazon VOD] MFA, DCQ or Captcha form
2022-02-18 21:55:19.616 T:11219   ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'NoneType' object has no attribute 'get_text'
                                                   Traceback (most recent call last):
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.amazon-test/default.py", line 5, in <module>
                                                       EntryPoint()
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.amazon-test/resources/lib/startup.py", line 77, in EntryPoint
                                                       exec('{}()'.format(mode))
                                                     File "<string>", line 1, in <module>
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.amazon-test/resources/lib/network.py", line 615, in LogIn
                                                       br = _MFACheck(br, email, soup)
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.amazon-test/resources/lib/network.py", line 484, in _MFACheck
                                                       msg = soup.find('span', attrs={'class': 'a-size-medium transaction-approval-word-break a-text-bold'}).get_text(strip=True)
                                                   AttributeError: 'NoneType' object has no attribute 'get_text'
                                                   -->End of Python script error report<--
```
Tried this 3 times, unlikely to be a email/password/captcha typo

**Comment by Sandmann79** [comment]
@ktaragorn this has been fixed with beta 3: [plugin.video.amazon-test-0.9.5_beta3.zip](https://github.com/Sandmann79/xbmc/files/8101641/plugin.video.amazon-test-0.9.5_beta3.zip)

**Comment by peakej** [comment]
Beta 3 has allowed me to sign in now.  But if I try to run the add-on it still fails. An error has occurred.

**Comment by Sandmann79** [comment]
@peakej your account is UK based?

**Comment by peakej** [comment]
> @peakej your account is UK based?

Yes, UK

**Comment by ktaragorn** [comment]
I am able to login, but dont seem to be able to get any video playing.. (I am not sure if i should be creating a seperate issue)

Logs look like this for a video play

```
2022-02-20 20:51:19.063 T:7328     INFO <general>: CActiveAESink::OpenSink - initialize sink
2022-02-20 20:51:19.064 T:7328     INFO <general>: Trying to open: samplerate: 44100, channelMask: 12, encoding: 4
2022-02-20 20:51:19.069 T:7328     INFO <general>: CAESinkAUDIOTRACK::Initializing with: m_sampleRate: 44100 format: AE_FMT_FLOAT (AE) method: PCM stream-type: PCM-STREAM min_buffer_size: 64656 m_frames: 2020 m_frameSize: 8 channels: 2
2022-02-20 20:51:20.050 T:8052    ERROR <general>: DoWork - Direct texture file loading failed for resource://resource.images.studios.white/Entertainment Film Distributors.png
2022-02-20 20:51:20.769 T:8053     INFO <general>: initializing python engine.
2022-02-20 20:51:24.873 T:8053     INFO <general>: [Amazon VOD] postURL: http://127.0.0.1:36422/gpr/https%3A%2F%2Fatv-ps-eu.amazon.co.uk%2Fcdp%2Fcatalog%2FGetPlaybackResources%3Fasin%3DB09QLN292V%26deviceTypeID%3DAOAGZA014O5RE%26firmware%3D1%26deviceID%3D4560e9221d8940b673a3407489faff9bd4ee54aaf054ba69853b5b1e%26marketplaceID%3DA1F83G8C2ARO7P%26format%3Djson%26version%3D1%26gascEnabled%3Dfalse%26subtitleFormat%3DTTMLv2%26operatingSystemName%3DWindows%26resourceUsage%3DImmediateConsumption%26consumptionType%3DStreaming%26deviceDrmOverride%3DCENC%26deviceStreamingTechnologyOverride%3DDASH%26deviceProtocolOverride%3DHttps%26deviceBitrateAdaptationsOverride%3DCVBR%252CCBR%26audioTrackId%3Dall%26languageFeature%3DMLFv2%26videoMaterialType%3DFeature%26desiredResources%3DPlaybackUrls%2CSubtitleUrls%2CForcedNarratives%2CTransitionTimecodes
2022-02-20 20:51:25.419 T:8053     INFO <general>: [Amazon VOD] Using Host: Cloudfront
2022-02-20 20:51:25.422 T:8053     INFO <general>: [Amazon VOD] checkURL: https://pop-dub-ww.cf.dash.row.aiv-cdn.net/dm/2$9meGYVa7-98KGmTTGH4J0MGRHMQ~/31af/0a8b/efd9/4d82-a859-5fe3cfaee082/ab9ca03d-06d3-4208-94ad-946ce2526380_corrected.mpd
2022-02-20 20:51:26.581 T:8053     INFO <general>: [Amazon VOD] Using inputstream.adaptive Version: 19.0.3
2022-02-20 20:51:26.749 T:7319     INFO <general>: VideoPlayer::OpenFile: plugin://plugin.video.amazon-test/?asin=B09QLN292V&mode=PlayVideo&name=My+Son&adult=0&trailer=0&selbitrate=0
2022-02-20 20:51:26.822 T:8009     INFO <general>: ADDON: Dll Destroyed - InputStream Adaptive
2022-02-20 20:51:26.824 T:8009     INFO <general>: Creating InputStream
2022-02-20 20:51:26.835 T:8053     INFO <general>: [Amazon VOD] Starting Playback: 62.56900405883789/6160.00048828125
2022-02-20 20:51:26.852 T:8009     INFO <general>: AddOnLog: inputstream.adaptive: SetVideoResolution (1920 x 1080)
2022-02-20 20:51:26.859 T:8009    ERROR <general>: GetDirectory - Error getting /data/user/0/org.xbmc.kodi/cache/apk/assets/addons/inputstream.adaptive/
2022-02-20 20:51:26.891 T:8009    ERROR <general>: AddOnLog: inputstream.adaptive: Key system request: com.widevine.alpha
2022-02-20 20:51:36.493 T:8009     INFO <general>: AddOnLog: inputstream.adaptive: Successfully parsed manifest file. #Periods: 1, #Streams in first period: 22, Type: VOD, Download speed: 0.0000 Bytes/s
2022-02-20 20:51:36.753 T:8009     INFO <general>: AddOnLog: inputstream.adaptive: Empty Service Certificate
2022-02-20 20:51:37.024 T:8009    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to find license;hdcpEnforcementResolutionPixels in JSON string
2022-02-20 20:51:37.111 T:8009     INFO <general>: AddOnLog: inputstream.adaptive: Empty Service Certificate
2022-02-20 20:51:37.397 T:8009    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to find license;hdcpEnforcementResolutionPixels in JSON string
2022-02-20 20:51:37.484 T:8009     INFO <general>: AddOnLog: inputstream.adaptive: Empty Service Certificate
2022-02-20 20:51:37.751 T:8009    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to find license;hdcpEnforcementResolutionPixels in JSON string
2022-02-20 20:51:37.759 T:8009     INFO <general>: Creating Demuxer
2022-02-20 20:51:37.764 T:8009     INFO <general>: Opening stream: 1001 source: 256
2022-02-20 20:51:37.899 T:8009     INFO <general>: Creating video codec with codec id: 27
2022-02-20 20:51:37.899 T:8009    ERROR <general>: CDVDVideoCodecAndroidMediaCodec::Open - InstanceGuard locked
2022-02-20 20:51:37.900 T:8009     INFO <general>: CVideoPlayerVideo::OpenStream - could not open video codec
2022-02-20 20:51:37.901 T:8009     INFO <general>: Opening stream: 1002 source: 256
2022-02-20 20:51:37.926 T:8009     INFO <general>: Finding audio codec for: 86056
2022-02-20 20:51:37.941 T:8009     INFO <general>: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder eac3
2022-02-20 20:51:37.942 T:8009     INFO <general>: CDVDAudioCodecAndroidMediaCodec Use raw decoder and decode using ff-eac3
2022-02-20 20:51:37.948 T:8009     INFO <general>: CDVDAudioCodecAndroidMediaCodec Open Android MediaCodec amc-eac3
2022-02-20 20:51:37.948 T:8009     INFO <general>: Closing stream player 3
2022-02-20 20:51:37.952 T:8009     INFO <general>: Opening stream: 0 source: 1025
2022-02-20 20:51:38.009 T:8018     INFO <general>: Creating video codec with codec id: 27
2022-02-20 20:51:38.013 T:8018     INFO <general>: CDVDVideoCodecAndroidMediaCodec::Open: Secure decoder requested: false (stream flags: 0)
2022-02-20 20:51:38.013 T:8018     INFO <general>: CDVDVideoCodecAndroidMediaCodec::Open Testing codec:OMX.amlogic.hevc.decoder.awesome
2022-02-20 20:51:38.014 T:8018     INFO <general>: CDVDVideoCodecAndroidMediaCodec::Open Testing codec:OMX.amlogic.avc.decoder.awesome
2022-02-20 20:51:38.112 T:8018     INFO <general>: CDVDVideoCodecAndroidMediaCodec:: Open Android MediaCodec OMX.amlogic.avc.decoder.awesome
2022-02-20 20:51:38.985 T:8018    ERROR <general>: CDVDVideoCodecAndroidMediaCodec::AddData error
2022-02-20 20:51:38.988 T:8027    ERROR <general>: CDVDAudioCodecAndroidMediaCodec::Decode ExceptionCheck
2022-02-20 20:51:38.997 T:8018    ERROR <general>: CDVDVideoCodecAndroidMediaCodec::AddData error
2022-02-20 20:51:39.001 T:8027    ERROR <general>: CDVDAudioCodecAndroidMediaCodec::Decode ExceptionCheck
2022-02-20 20:51:39.010 T:8018    ERROR <general>: CDVDVideoCodecAndroidMediaCodec::AddData error
2022-02-20 20:51:39.014 T:8027    ERROR <general>: CDVDAudioCodecAndroidMediaCodec::Decode ExceptionCheck
2022-02-20 20:51:39.024 T:8018    ERROR <general>: CDVDVideoCodecAndroidMediaCodec::AddData error
2022-02-20 20:51:39.028 T:8027    ERROR <general>: CDVDAudioCodecAndroidMediaCodec::Decode ExceptionCheck
2022-02-20 20:51:39.036 T:8018    ERROR <general>: CDVDVideoCodecAndroidMediaCodec::AddData error
```

**Comment by Sandmann79** [comment]
@ktaragorn this line points out a widevine related error:
`2022-02-20 20:51:37.751 T:8009    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to find license;hdcpEnforcementResolutionPixels in JSON string`
Since you have a android device you have to check if your it supports Widevine L1. 
If yes, you maybe have to upgrade your operating system to a newer version.

As far as i remember without Widevine L1 support on your device, there's no chance to get playback working.

**Comment by vinceducat** [comment]
the probleme seems to be with Widevine version... many add ons have this probleme at this time...

**Comment by Sandmann79** [comment]
Has been fixed with 69da66d2949ae6bf3b3e63c33ffe0ff4012de242
